### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by wolterbwambale**
### In this project, i have implemented the following:-
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      user = User.create(
+        name: 'test user',
+        photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+        bio: 'test_bio',
+        posts_counter: 0
+      )
+
+      expect(user.recent_posts).to be_empty
+    end
+  end
```
Comment (jerryowusu) [inline spec/model/user_spec.rb]: Good job so far. However, I can see that your test here is mostly for the recent_posts method. 
Kindly add unit tests to `validate the presence of name`. 
You should also consider adding a test to check that `posts_counter must be an integer greater than or equal to zero`. :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @wolterbwambale , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so-close


## Highlights
Neat work so far :+1: 
All written tests passed :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wolterbwambale** [comment]
@jerryowusu  thank you, i have fixed the required changes

**Review by Gambit142 — APPROVED** [review]
Hi @wolterbwambale 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Setup linters correctly ✔️ 
- Use of Rspec to test models ✔️ 
- Wrote validations for model attributes ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
